sub-packages/bionemo-evo2/src/bionemo/evo2/utils/logging/callbacks.py

```diff
@@ -43,6 +43,20 @@ class TEVCallback(Callback):
     - Data Parallelism: Only logs on rank 0 of each model parallel group
     """
 
+    def on_validation_start(self, trainer, pl_module) -> None:
+        """Clean up CUDA memory before validation to prevent initialization errors."""
+        if torch.cuda.is_available():
+            try:
+                torch.cuda.empty_cache()
+                torch.cuda.synchronize()
+                current_device = torch.cuda.current_device()
+                torch.cuda.set_device(current_device)
+                torch.cuda.synchronize()
+                import gc
+                gc.collect()
+            except Exception as e:
+                print(f"Warning: CUDA cleanup failed: {e}")
+
```

Comment:
I'm ok with this but can you make a new purpose built callback called something like `GarbageCollectAtInferenceTime` or something that just does this? Then add it as a new callback to the trainer?